Library/Context/ContextStorageService.php

```diff
@@ -282,5 +282,83 @@ public function updateWorkflowStepJob(string $workflowProcessId, string $stepCod
+        string $workflowProcessId,
+        string $stepCode,
+        int $jobId,
+        string $workerHostname,
+        string $workerCode
+    ) {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $script = <<<eot
+        def nbStep = ctx._source.steps.size();
+        def jobHash = [reservedBy:workerCode, executedOn:workerHostname];
+        // loop over steps
+        for (stepIndex = 0; stepIndex < nbStep; stepIndex++) {
+            def step = ctx._source.steps[stepIndex];
+            // if its a parrallelized steps node, loop over each
+            if (step instanceof List) {
+                def nbParallelizedStep = step.size();
+                for (parallelizedStepIndex= 0; parallelizedStepIndex< nbParallelizedStep; parallelizedStepIndex++) {
+                    // if the given step is found, look for the given job
+                    if (step[parallelizedStepIndex].name == stepCode) {
+                        def nbJob = step[parallelizedStepIndex]['jobList'].size();
+                        for (jobIndex = 0; jobIndex < nbJob; jobIndex++) {
+                            def job = step[parallelizedStepIndex]['jobList'][jobIndex];
+                            if (job.id == jobId) {
+                                // if job's already reserved : noop
+                                if (job.containsKey('reservedBy')) {
+                                    ctx.op = 'noop';
+                                    break;
+                                }
+                                ctx._source.steps[stepIndex][parallelizedStepIndex]['jobList'][jobIndex] << jobHash
+                                break;
+                            }
+                         }
+                        break;
+                    }
+                }
+            } else if (step.name == stepCode) {
+                def nbJob = step.jobList.size();
+                for (jobIndex = 0; jobIndex < nbJob; jobIndex++) {
+                    def job = ctx._source.steps[stepIndex]['jobList'][jobIndex];
+                    if (job.id == jobId) {
+                        // if job's already reserved : noop
+                        if (job.containsKey('reservedBy')) {
+                            ctx.op = 'noop';
+                            break;
+                        }
+                        ctx._source.steps[stepIndex]['jobList'][jobIndex] << jobHash
+                        break;
+                    }
+                }
+            }
+        }
+eot;
+        $updateHash = [
+            'script' => [
+                'lang' => 'groovy',
+                'inline' => $script,
+                'params' => [
+                    'stepCode' => $stepCode,
+                    'jobId' => $jobId,
+                    'workerHostname' => $workerHostname,
+                    'workerCode' => $workerCode
+                ]
+            ]
+        ];
+        return $this->adapter->update($workflowProcessId, $updateHash, 2);
```

Comment:
2 in a const?

Reply:
indeed, right, even in conf, btw, it stands for the number of els update retries in case of update conflicts

Reply:
I'll add a ticket to introduce a shared DisturbConfig object, there are some todos related to this...
will also add a todo for this one